### ISSUE TITLE
Option to remove "sourcesContent" from generated source map file?

### ISSUE THREAD

**Issue by umairkhan-dev** [javascript-obfuscator/javascript-obfuscator#956]
Hi, I am obfuscating JS code using **javascript-obfuscator** CLI with `source map` enabled. The generated `map` file contains `sourcesContent` which in turn contains all the source of obfuscated code. Is there an option to tell **javascript-obfuscator** not to include source content in `map` file.

The CLI command I am using.
```
javascript-obfuscator themes/blue/.source/js --output themes/blue/dist/js --compact true --self-defending true --source-map true
```

Thanks.

**Comment by sanex3339** [comment]
For correct source map work, it should contain one of two fields: 
1) `sourcesContent` field with original source code content. 
2) `sources` field that will be used by the browser to download the original source code for source map remapping.

**Comment by umairkhan-dev** [comment]
I know. I do not have any problem with `source`. Just want to remove `sourcesContent`.

The scenario is that my source file is in separate directory, which I do not upload. The **dev tools** may suggest me the line number in case of any log or error, which I can view from source file. However as **dev tools** can not access source file, the browser can not convert obfuscated code to normal code. But with `sourcesContent`, the browser does not need source file to convert obfuscated code to normal code. It can just read normal code from `map` file.

Hopefully, I am able to explain my scenario.

**Comment by da411d** [comment]
@umairkhan-dev try to replace all `sourcesContent` strings with random data.

**Before:**
```
{
  "version": 3,
  "sources": [
    "sourceMap"
  ],
  "names": [
    "hi",
    "console"
  ],
  "mappings": "ohRACA,SAASA,EAAT,E,+05CACEC,OAAA,C,yCAAA,E,mDAAA,C,4yHAEFD,EAAA,G",
  "sourcesContent": [
    "// Paste your JavaScript code here\nfunction hi() {\n  console.log(\"Hello World!\");\n}\nhi();"
  ]
}
```
![image](https://user-images.githubusercontent.com/12462454/126082034-932ca2bc-b91e-45f5-8254-082a3fdfb94c.png)


**After:**
```
{
   "version": 3,
   "sources": [
      "sourceMap"
   ],
   "names": [
      "hi",
      "console"
   ],
   "mappings": "0tFACA,SAASA,EAAT,E,6ngBACEC,OAAA,C,gFAAA,E,sFAAA,C,8tKAEFD,EAAA",
   "sourcesContent": [
      "// uu9p6 se6i kqkxw0eb51 1wfl 3mtz\n3csyckp4 q2() {\n  odmmfs0.71f(\"a2l6u lsx8x!\");\n}\nxl();"
   ]
}
```
![image](https://user-images.githubusercontent.com/12462454/126082139-9e617605-b81e-4154-b331-ec8fdc2a9c1d.png)


**My code:**
```
const randomLetter = () => Math.random().toString(36).substr(2, 1);
const map = {...}; // Just copied sourcemap file content
map.sourcesContent = map.sourcesContent.map(row => row.replace(/\w/ig, randomLetter));
```

**Comment by sanex3339** [comment]
I'm not sure the best way to fix this.
Right now all source code is placed into the `sourcesContent` field of `source map`.

I can add a new option `sourceMapContent` or `sourceMapSourcesMode` with 2 values:
- `sourcesContent` - default, current behaviour. When enabled adds `sources` field with fake  source named `sourceMap` and adds `sourcesContent` field with the original source code
- `sources` - When enabled - only adds `sources` field with source description of original source code file

**Comment by umairkhan-dev** [comment]
> I can add a new option `sourceMapSourcesMode` with 2 values:
> 
>     * `sourcesContent` - default, current behaviour. When enabled adds `sources` field with fake  source named `sourceMap` and adds `sourcesContent` field with the original source code
> 
>     * `sources` - When enabled - only adds `sources` field with source description of original source code file

@sanex3339 May I suggest:
```
javascript-obfuscator .source/js --output js --source-map true --source-map-include-sources false
```

### `sourceMapIncludeSources`
Type: `boolean` Default: `false`

##### :warning: This option forcibly enables `sourceMap` option.

Whether you want to include the content of source files in the source map as `sourcesContent` property or not.

**Comment by da411d** [comment]
Okay, I didn't think that you can just remove `sourcesContent` field

**Comment by umairkhan-dev** [comment]
@da411d Yup, For the time being, I am manually removing / deleting `sourcesContent` property from `map` file.

**Comment by sanex3339** [comment]
I personally prefer my variant with `sourceMapSourcesMode` option. Will implement it this week.

**Comment by sanex3339** [comment]
https://github.com/javascript-obfuscator/javascript-obfuscator/pull/957

**Comment by sanex3339** [comment]
Released as `2.17.0`.
For CLI use new option `--source-map-sources-mode sources`

**Comment by sanex3339** [comment]
Feel free to ask about the new option here or create a new issue.
